eng/pipelines/smoke-tests.yml

```diff
@@ -67,46 +67,46 @@ jobs:
       - pwsh: dotnet restore ./common/SmokeTests/SmokeTest/SmokeTest.csproj
         displayName: dotnet restore
 
-      # Set secret environment variables for different clouds
-      - pwsh: |
-          $variables = @{
-            KEY_VAULT_URI='$(smoke-tests-key-vault-project-url)'
-            EVENT_HUBS_CONNECTION_STRING='$(smoke-tests-event-hubs-connection-string)'
-            BLOB_CONNECTION_STRING='$(smoke-tests-storage-connection-string)'
-            AZURE_CLIENT_SECRET='$(aad-azure-sdk-test-client-secret)'
-            AZURE_TENANT_ID='$(aad-azure-sdk-test-tenant-id)'
-            AZURE_CLIENT_ID='$(aad-azure-sdk-test-client-id)'
-            AZURE_AUTHORITY_HOST='$(aad-azure-sdk-test-authority-uri)'
-            COSMOS_AUTH_KEY='$(smoke-tests-cosmos-key)'
-            COSMOS_URI='$(smoke-tests-cosmos-endpoint)'
-          };
-          foreach ($key in $variables.Keys) {
-            Write-Host "Setting variable '$key'"
-            Write-Host "##vso[task.setvariable variable=_$key;issecret=true;]$($variables[$key])"
-            Write-Host "##vso[task.setvariable variable=$key;]$($variables[$key])"
-          }
-        displayName: Set secrets for public cloud
-        condition: and(succeeded(), eq(variables['CloudType'], 'public'))
+      - template: ../common/TestResources/deploy-test-resources.yml
+        parameters:
+          ServiceDirectory: '$(Build.SourcesDirectory)/common/SmokeTests/'
```

Comment:
Is there still a goal to reduce these further? Also at what point should we simply turn these into matrix parameters instead of conditioning based on CloudType?